packages/SwingSet/test/upgrade/test-upgrade.js

```diff
@@ -537,6 +546,95 @@ test('non-durable exports are abandoned by vat upgrade', async t => {
   );
 });
 
+test('non-durable exports are abandoned by upgrade of non-liveslots vat', async t => {
+  const config = makeConfigFromPaths('../bootstrap-relay.js', {
+    defaultManagerType: 'xs-worker',
+  });
+  config.vats.exporter = {
+    sourceSpec: bfile('../vat-direct.js'),
+    parameters: { vatName: 'exporter' },
+    creationOptions: { enableSetup: true },
+  };
+  config.vats.observer = {
+    sourceSpec: bfile('../vat-direct.js'),
+    parameters: { vatName: 'observer' },
+    creationOptions: { enableSetup: true },
+  };
+  const { controller, kvStore, run } = await initKernelForTest(
+    t,
+    t.context.data,
+    config,
+    { holdObjectRefs: false },
+  );
+
+  const exporterVatID = controller.vatNameToID('exporter');
+
+  // Export two objects from exporter to observer,
+  // one to be held strongly and the other weakly.
+  const observerPresence = await run('getVatRoot', [{ name: 'observer' }]);
+  const observer = await run('awaitVatObject', [
+    { presence: observerPresence, rawOutput: true },
+  ]);
+  const strongObj = await run('exportFakeObject', [], 'exporter');
+  await run(
+    'syscall-send',
+    [observer, 'acceptImports', [strongObj]],
+    'exporter',
+  );
+  const weakObj = await run('exportFakeObject', [], 'exporter');
+  await run(
+    'syscall-send',
+    [observer, 'acceptWeakImports', [weakObj]],
+    'exporter',
+  );
+
+  // Verify kernel tracking of the objects.
+  const strongKref = krefOf(strongObj);
+  t.is(kvStore.get(`${strongKref}.owner`), exporterVatID);
+  t.is(
+    kvStore.get(`${strongKref}.refCount`),
+    '1,1',
+    'strong observation must increment both reachable and recognizable ref counts',
+  );
+  const weakKref = krefOf(weakObj);
+  t.is(kvStore.get(`${weakKref}.owner`), exporterVatID);
+  t.is(
+    kvStore.get(`${weakKref}.refCount`),
+    '0,1',
+    'weak observation must increment recognizable but not reachable ref counts',
+  );
+
+  // Null-upgrade exporter and verify updated kernel tracking.
+  const bundle = await bundleSource(config.vats.exporter.sourceSpec);
+  const bundleID = await controller.validateAndInstallBundle(bundle);
+  const upgradeKpid = controller.upgradeStaticVat('exporter', false, bundleID, {
+    vatParameters: config.vats.exporter.parameters,
+  });
+  await controller.run();
+  t.is(controller.kpStatus(upgradeKpid), 'fulfilled');
+  t.is(kvStore.get(`${strongKref}.owner`), undefined);
+  t.is(kvStore.get(`${weakKref}.owner`), undefined);
+  t.is(
+    kvStore.get(`${strongKref}.refCount`),
+    '1,1',
+    'strong observation of abandoned object retains ref counts',
+  );
+  // TODO: Expect and verify observer receipt of dispatch.retireExports
+  // and corresponding removal of weakKref ref counts.
+  // https://github.com/Agoric/agoric-sdk/issues/7212
+  t.is(
+    kvStore.get(`${weakKref}.refCount`),
+    '0,1',
+    'weak observation of abandoned object retains ref counts before #7212',
```

Comment:
once this lands, please add a pointer to the file and line number in #7212, so we'll know which tests to update/check when we fix it